pkg/output/table_filter.go

```diff
@@ -0,0 +1,159 @@
+	var newHeaders []string
+	var columnsToKeep []int
+
+	for i, header := range headers {
+		if shouldKeepColumn(i) {
+			newHeaders = append(newHeaders, header)
+			columnsToKeep = append(columnsToKeep, i)
+		}
+	}
+
+	var newTableData [][]string
+	for _, row := range tableData {
+		var newRow []string
+		for _, col := range columnsToKeep {
+			if col < len(row) {
+				newRow = append(newRow, row[col])
+			} else {
+				newRow = append(newRow, "")
+			}
+		}
+		newTableData = append(newTableData, newRow)
+	}
+
+	return newHeaders, newTableData
+}
+
+func RemoveEmptyColumns(headers []string, tableData [][]string) [][]string {
+	if len(tableData) == 0 {
+		return [][]string{headers}
+	}
+
+	// Use FilterColumns with a predicate that keeps non-empty columns
+	newHeaders, newTableData := FilterColumns(headers, tableData, func(columnIdx int) bool {
+		return !CheckIfColumnIsEmpty(columnIdx, tableData)
+	})
+
+	// Combine headers and data as before for backward compatibility
+	var result [][]string
+	result = append(result, newHeaders)
+	result = append(result, newTableData...)
+
+	return result
+}
+
+func FilterColumnsWithConditions(headers []string, tableData [][]string, preserveColumn map[int]bool) ([]string, [][]string) {
+	return FilterColumns(headers, tableData, func(columnIdx int) bool {
+		// Keep column if it's marked to be preserved OR if it has data
+		return preserveColumn[columnIdx] || !CheckIfColumnIsEmpty(columnIdx, tableData)
+	})
+}
+
+// RemoveEmptyColumnsWithSeparators removes empty columns and filters the separators list accordingly
+func RemoveEmptyColumnsWithSeparators(headers []string, tableData [][]string, separators []string) ([][]string, []string) {
+	if len(tableData) == 0 {
+		return [][]string{headers}, separators
+	}
+
+	var newHeaders []string
+	var newSeparators []string
+	var columnsToKeep []int
+
+	for i, header := range headers {
+		if !CheckIfColumnIsEmpty(i, tableData) {
+			newHeaders = append(newHeaders, header)
+			columnsToKeep = append(columnsToKeep, i)
+			if i < len(separators) {
+				newSeparators = append(newSeparators, separators[i])
+			}
+		}
+	}
+
+	var result [][]string
+	result = append(result, newHeaders)
+
+	for _, row := range tableData {
+		var newRow []string
+		for _, col := range columnsToKeep {
+			if col < len(row) {
+				newRow = append(newRow, row[col])
+			} else {
+				newRow = append(newRow, "")
+			}
+		}
+		result = append(result, newRow)
+	}
+
+	return result, newSeparators
+}
+
+func BuildTable(writer *tabwriter.Writer, separator string, tableData [][]string) {
+	// Build separators list from single separator for backward compatibility
+	if len(tableData) == 0 {
+		return
+	}
+
+	maxCols := 0
+	for _, row := range tableData {
+		if len(row) > maxCols {
+			maxCols = len(row)
+		}
+	}
+
+	separators := make([]string, maxCols)
+	for i := range separators {
+		separators[i] = separator
+	}
+
+	BuildTableWithSeparators(writer, separators, tableData)
```

Comment:
https://github.com/openshift/rosa/pull/2983#discussion_r2325290550

In this comment I talked about having support for separators which have different separators at different parts of the table. For example, `val1\tval2\t\tval3`

It seems like you have support for it now, but did not actually support it, instead wrapping it in a function which makes a copy of a single separator that was passed in

If we could make that functional, and get the original separators back in the command which have unique separator values per column, that would be great

Reply:
Improved this on the last commit

Reply:
Worked on this on this commit https://github.com/openshift/rosa/pull/2983/commits/3f5390f0fcb7d52c619f600aa27320e2dfe4f7c0